$OpenBSD: patch-src_hotspot_cpu_aarch64_assembler_aarch64_cpp,v 1.1 2021/10/31 21:16:41 kurt Exp $

adrp implementation should match prototype in header.
Fix implicit conversion failures from:
	int64_t -> jlong.

Index: src/hotspot/cpu/aarch64/assembler_aarch64.cpp
--- src/hotspot/cpu/aarch64/assembler_aarch64.cpp.orig
+++ src/hotspot/cpu/aarch64/assembler_aarch64.cpp
@@ -1532,7 +1532,7 @@ void Address::lea(MacroAssembler *as, Register r) cons
   }
 }
 
-void Assembler::adrp(Register reg1, const Address &dest, uintptr_t &byte_offset) {
+void Assembler::adrp(Register reg1, const Address &dest, uint64_t &byte_offset) {
   ShouldNotReachHere();
 }
 
@@ -1702,7 +1702,7 @@ void Assembler::add_sub_immediate(Register Rd, Registe
 }
 
 bool Assembler::operand_valid_for_add_sub_immediate(int64_t imm) {
-  uint64_t uimm = (uint64_t)uabs(imm);
+  uint64_t uimm = (uint64_t)uabs((jlong)imm);
   if (uimm < (1 << 12))
     return true;
   if (uimm < (1 << 24)
